From fadb453930cd94af31bf662eac7df2528c2fc7f3 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Tue, 11 Aug 2020 18:38:42 -0700
Subject: [PATCH 1511/1851] misc: xilinx-ai-engine: use IS_ENABLED() compile
 guard
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The #ifdef doesn't catch the module build correctly, where CONFIG is
set with _MODULE. For module build, it causes the build error:

drivers/misc/xilinx-ai-engine/ai-engine-dev.c:517:6: error: redefinition of ‘aie_partition_is_available’
...

Use IS_ENABLED() to fix.

Reported-by: kernel test robot <lkp@intel.com>
Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Reviewed-by: Wendy Liang <wendy.liang@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 include/linux/xlnx-ai-engine.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/xlnx-ai-engine.h b/include/linux/xlnx-ai-engine.h
index d499665d412d..54b49ee6cfad 100644
--- a/include/linux/xlnx-ai-engine.h
+++ b/include/linux/xlnx-ai-engine.h
@@ -8,14 +8,14 @@
 #ifndef _XLNX_AI_ENGINE_H_
 #define _XLNX_AI_ENGINE_H_
 
-#ifndef CONFIG_XILINX_AIE
+#if !IS_ENABLED(CONFIG_XILINX_AIE)
 #include <linux/errno.h>
 #endif
 #include <uapi/linux/xlnx-ai-engine.h>
 
 struct device;
 
-#ifdef CONFIG_XILINX_AIE
+#if IS_ENABLED(CONFIG_XILINX_AIE)
 bool aie_partition_is_available(struct aie_partition_req *req);
 struct device *aie_partition_request(struct aie_partition_req *req);
 int aie_partition_get_fd(struct device *dev);
-- 
2.31.1

